lib/src/zap/sock/zap_sock.c

```diff
@@ -1161,14 +1161,16 @@ static void sock_write(ovis_event_t ev)
 
 	/* msg */
 	while (wr->msg_len) {
-		wsz = write(sep->sock, wr->msg + wr->off, wr->msg_len);
+		wsz = send(sep->sock, wr->msg + wr->off, wr->msg_len, MSG_NOSIGNAL);
 		if (wsz < 0) {
 			if (errno == EAGAIN || errno == EWOULDBLOCK) {
 				__enable_epoll_out(sep);
 				goto out;
 			}
+			if (errno == EPIPE)
```

Comment:
EPIPE is telling you it's EOF, so why are you skipping the shutdown call?

Reply:
Because I think that `shutdown()` will eventually be called when we handle EPOLLHUP. Calling `shutdown()` on an disconnected socket will just return ENOTCONN. Shall I change the code to call `shutdown()` in this case too?